policy/release/tasks.rego

```diff
@@ -0,0 +1,54 @@
+# METADATA
+# custom:
+#   tasks_required:
+#     rule_data:
+#       required_task_refs:
+#       - add-sbom-and-push
+#       - clamav-scan
+#       - deprecated-image-check
+#       - get-clair-scan
+#       - sanity-inspect-image
+#       - sanity-label-check
+#       - sanity-optional-label-check
+#       - sast-go
+package policy.release.tasks
+
+import data.lib
+import future.keywords.in
+
+all_tasks_missing_errors = {e | task := rego.metadata.chain()[_].annotations.custom.tasks_required.rule_data.required_task_refs[_]; e := {
+	"code": "tasks_required",
+	"msg": sprintf("Required task '%s' not found in the PipelineRun attestation", [task]),
+	"effective_on": "2022-01-01T00:00:00Z",
+}}
+
+# METADATA
+# title: No tasks run
+# description: |-
+#   This policy enforces that at least one Task is preset in the PipelineRun
+#   attestation.
+# custom:
+#   short_name: tasks_missing
+#   failure_msg: No tasks found in PipelineRun attestation
+deny[result] {
+	att := lib.pipelinerun_attestations[_]
+	count(att.predicate.buildConfig.tasks) == 0
+	result := lib.result_helper(rego.metadata.chain(), [])
+}
+
+# METADATA
+# title: Required tasks not run
+# description: |-
+#   This policy enforces that the required set of tasks is run in a
+#   PipelineRun.
+# custom:
+#   short_name: tasks_required
+#   failure_msg: Required task '%s' not found in the PipelineRun attestation
+deny[result] {
+	att := lib.pipelinerun_attestations[_]
+	attested_tasks := {t | t := att.predicate.buildConfig.tasks[_].name}
+	required_tasks := {t | t := rego.metadata.chain()[_].annotations.custom.tasks_required.rule_data.required_task_refs[_]}
+	all_missing := required_tasks - attested_tasks
+	missing := all_missing[_]
```

Comment:
Will this cause it to emit multiple results? Can we add a test to illustrate what happens when more than one tasks are missing? There's `test_no_tasks_present` but that's asserting against a dynamic value, `all_tasks_missing_errors`. Can we add a test that asserts against a literal so the intended behavior is more obvious?

Reply:
In `test_all_but_one_required_task_not_present` all but one task is missing, added `test_several_tasks_not_present` where all but three tasks are missing.